src/services/codefixes/importFixes.ts

```diff
@@ -620,4 +658,74 @@ namespace ts.codefix {
         // Need `|| "_"` to ensure result isn't empty.
         return !isStringANonContextualKeyword(res) ? res || "_" : `_${res}`;
     }
+
+    function createLazyPackageJsonDependencyReader(fromFile: SourceFile, host: LanguageServiceHost) {
+        const packageJsonPaths = findPackageJsons(getDirectoryPath(fromFile.fileName), host);
+        const dependencyIterator = readPackageJsonDependencies();
+        let seenDeps: Map<true> | undefined;
+        function *readPackageJsonDependencies() {
+            type PackageJson = Record<typeof dependencyKeys[number], Record<string, string> | undefined>;
+            const dependencyKeys = ["dependencies", "devDependencies", "optionalDependencies"] as const;

```

Comment:
I saw string completions consider `peerDependencies` here, but I was on the fence about whether that makes sense here. Usually a peer dependency is also a dev dependency, if it’s not, we probably don’t have typings for it anyway, and would never offer an auto-import in the first place. It seems unlikely to me that anyone will notice/care whether we look there.

Reply:
Yeah, I think excluding peer deps is the right thing to do here, if it's not in your dependencies (or through other deps) then it won't be in the tree at all